packages/nimble-components/src/wafer-map/modules/prerendering.ts

```diff
@@ -131,13 +132,27 @@ export class Prerendering {
     }
 
     private calculateOpacity(
-        selectedValue: string,
-        highlightedValues: Readonly<string[]>
+        dieTags?: string[],
+        highlightedTags?: string[]
     ): number {
-        return highlightedValues.length > 0
-            && !highlightedValues.some(dieValue => dieValue === selectedValue)
-            ? this.nonHighlightedOpacity
-            : 1;
+        if (
+            highlightedTags?.length === 1
+            && highlightedTags?.every(tag => tag === '')
+        ) {
+            return this.nonHighlightedOpacity;
+        }
+        const tagsMatch = this.isAnyDieTagPresent(dieTags, highlightedTags);
+        return tagsMatch ? this.nonHighlightedOpacity : 1;
+    }
+
+    private isAnyDieTagPresent(
```

Comment:
I think the logic from here can be combined with the method above without issue, as that method just does a empty check before calling this